web-api/src/mind.rs

```diff
@@ -209,6 +259,40 @@ impl DocumentProvider {
     fn get(&self, id: &DocumentId) -> Option<&Document> {
         self.documents.get(id)
     }
+
+    // get all documents that matches user's interest
+    fn get_all_interest(&self, interests: &[String]) -> Vec<&Document> {
+        self.documents
+            .values()
+            .filter(|doc| doc.is_interesting(interests))
+            .collect()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct Users(HashMap<UserId, Vec<String>>);
```

Comment:
:art: if we also derive `Deref` for `Users`, then we can forward method calls directly to the inner type, for example instead of `users.0.iter()` we can just call `user.iter()`. this is not available in the `std` lib though, but the `derive_more` crate got this covered (it's already a dependency of our crate).
```suggestion
#[derive(Debug, derive_more::Deref, Deserialize)]
struct Users(HashMap<UserId, Vec<String>>);
```

Reply:
That's a nice feature, I will apply it